src/CurlTransport.php

```diff
@@ -141,12 +141,10 @@ private function prepare($request)
         }
 
         $content = $request->getContent();
-        if ($content === null) {
-            if ($method === 'HEAD') {
-                $curlOptions[CURLOPT_NOBODY] = true;
-            }
-        } else {
-            $curlOptions[CURLOPT_POSTFIELDS] = $content;
+        $curlOptions[CURLOPT_POSTFIELDS] = $content;
```

Comment:
Looks like it is filling POSTFIELDS for HEAD request but HEAD should have no body.

Reply:
Yes, but I think it's OK as long as it's null. Do you think something will break? Doesn't CURL handle this correctly if user sets body for HEAD request, does it?

Reply:
I think we should behave [according to specification](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html):

> The HEAD method is identical to GET except that the server MUST NOT return a message-body in the response.

So no, it's not OK to set POSTFIELDS in case of HEAD request.

Reply:
That quote from spec says about a response, not request. But I see your point, I'll change it.

Reply:
"identical to GET" means no body.